nixos/modules/system/boot/loader/bootloaders.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  cfg = config.boot.loader;
+  foldEnabledBootloaders = fn: start:
+    foldAttrs fn start
+    # Reverse order is a conscious choice to make it so that "last append" is first considered here.
+    # by default.
+      (map (moduleId: cfg.builders.${moduleId}) (reverseList cfg.enabled));
+in
+{
+  # Interface
+  options = {
+    boot.loader.builders = mkOption {
+      type = types.attrsOf (types.submodule ({ ... }: {
+        options = {
+          id = mkOption {
+            type = types.str;
+            description = lib.mdDoc "A string describing the bootloader name";
+          };
+          script = mkOption {
+            type = types.package;
+            description = lib.mdDoc "An executable script that ...";
+          };
+        };
+      }));
+      default = {};
+      example = literalExpression ''{ systemd-boot = finalSystemBootBuilder; }'';
+      internal = true;
+      description = lib.mdDoc ''
+        An attribute set of builder for any potential bootloader.
+        This is set internally by each bootloader with their
+        final system boot builder script.
+      '';
+    };
```

Comment:
More of a stylistic concern, can we move the submodule declaration into the `let` attribute. to make the actual option easier to parse visually?

Right now it took me a small while to realise that `boot.loader.builders` did not have options `id` and `script`, but was composed of submodules.

Reply:
For sure

Reply:
Let me know if you feel like this is still a concern in the current model.